// build_io_api_v1/dist/model/ApiV1OidcLoginGet200Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ApiV1OidcLoginGet200ResponseSpec = _interopRequireDefault(require("./ApiV1OidcLoginGet200ResponseSpec"));
var _ApiV1OidcLoginGet200ResponseStatus = _interopRequireDefault(require("./ApiV1OidcLoginGet200ResponseStatus"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Build.io API V1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ApiV1OidcLoginGet200Response model module.
 * @module model/ApiV1OidcLoginGet200Response
 * @version v1
 */
var ApiV1OidcLoginGet200Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApiV1OidcLoginGet200Response</code>.
   * @alias module:model/ApiV1OidcLoginGet200Response
   * @param kind {String} 
   * @param apiVersion {String} 
   * @param spec {module:model/ApiV1OidcLoginGet200ResponseSpec} 
   * @param status {module:model/ApiV1OidcLoginGet200ResponseStatus} 
   */
  function ApiV1OidcLoginGet200Response(kind, apiVersion, spec, status) {
    _classCallCheck(this, ApiV1OidcLoginGet200Response);
    ApiV1OidcLoginGet200Response.initialize(this, kind, apiVersion, spec, status);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ApiV1OidcLoginGet200Response, null, [{
    key: "initialize",
    value: function initialize(obj, kind, apiVersion, spec, status) {
      obj['kind'] = kind;
      obj['apiVersion'] = apiVersion;
      obj['spec'] = spec;
      obj['status'] = status;
    }

    /**
     * Constructs a <code>ApiV1OidcLoginGet200Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApiV1OidcLoginGet200Response} obj Optional instance to populate.
     * @return {module:model/ApiV1OidcLoginGet200Response} The populated <code>ApiV1OidcLoginGet200Response</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApiV1OidcLoginGet200Response();
        if (data.hasOwnProperty('kind')) {
          obj['kind'] = _ApiClient["default"].convertToType(data['kind'], 'String');
        }
        if (data.hasOwnProperty('apiVersion')) {
          obj['apiVersion'] = _ApiClient["default"].convertToType(data['apiVersion'], 'String');
        }
        if (data.hasOwnProperty('spec')) {
          obj['spec'] = _ApiV1OidcLoginGet200ResponseSpec["default"].constructFromObject(data['spec']);
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiV1OidcLoginGet200ResponseStatus["default"].constructFromObject(data['status']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ApiV1OidcLoginGet200Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ApiV1OidcLoginGet200Response</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ApiV1OidcLoginGet200Response.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['kind'] && !(typeof data['kind'] === 'string' || data['kind'] instanceof String)) {
        throw new Error("Expected the field `kind` to be a primitive type in the JSON string but got " + data['kind']);
      }
      // ensure the json data is a string
      if (data['apiVersion'] && !(typeof data['apiVersion'] === 'string' || data['apiVersion'] instanceof String)) {
        throw new Error("Expected the field `apiVersion` to be a primitive type in the JSON string but got " + data['apiVersion']);
      }
      // validate the optional field `spec`
      if (data['spec']) {
        // data not null
        _ApiV1OidcLoginGet200ResponseSpec["default"].validateJSON(data['spec']);
      }
      // validate the optional field `status`
      if (data['status']) {
        // data not null
        _ApiV1OidcLoginGet200ResponseStatus["default"].validateJSON(data['status']);
      }
      return true;
    }
  }]);
}();
ApiV1OidcLoginGet200Response.RequiredProperties = ["kind", "apiVersion", "spec", "status"];

/**
 * @member {String} kind
 */
ApiV1OidcLoginGet200Response.prototype['kind'] = undefined;

/**
 * @member {String} apiVersion
 */
ApiV1OidcLoginGet200Response.prototype['apiVersion'] = undefined;

/**
 * @member {module:model/ApiV1OidcLoginGet200ResponseSpec} spec
 */
ApiV1OidcLoginGet200Response.prototype['spec'] = undefined;

/**
 * @member {module:model/ApiV1OidcLoginGet200ResponseStatus} status
 */
ApiV1OidcLoginGet200Response.prototype['status'] = undefined;
var _default = exports["default"] = ApiV1OidcLoginGet200Response;
