// build_io_api_v1/dist/model/ApiV1AppsIdBuildsPostRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Build.io API V1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ApiV1AppsIdBuildsPostRequest model module.
 * @module model/ApiV1AppsIdBuildsPostRequest
 * @version v1
 */
var ApiV1AppsIdBuildsPostRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApiV1AppsIdBuildsPostRequest</code>.
   * @alias module:model/ApiV1AppsIdBuildsPostRequest
   */
  function ApiV1AppsIdBuildsPostRequest() {
    _classCallCheck(this, ApiV1AppsIdBuildsPostRequest);
    ApiV1AppsIdBuildsPostRequest.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ApiV1AppsIdBuildsPostRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ApiV1AppsIdBuildsPostRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApiV1AppsIdBuildsPostRequest} obj Optional instance to populate.
     * @return {module:model/ApiV1AppsIdBuildsPostRequest} The populated <code>ApiV1AppsIdBuildsPostRequest</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApiV1AppsIdBuildsPostRequest();
        if (data.hasOwnProperty('branch')) {
          obj['branch'] = _ApiClient["default"].convertToType(data['branch'], 'String');
        }
        if (data.hasOwnProperty('commitish')) {
          obj['commitish'] = _ApiClient["default"].convertToType(data['commitish'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ApiV1AppsIdBuildsPostRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ApiV1AppsIdBuildsPostRequest</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['branch'] && !(typeof data['branch'] === 'string' || data['branch'] instanceof String)) {
        throw new Error("Expected the field `branch` to be a primitive type in the JSON string but got " + data['branch']);
      }
      // ensure the json data is a string
      if (data['commitish'] && !(typeof data['commitish'] === 'string' || data['commitish'] instanceof String)) {
        throw new Error("Expected the field `commitish` to be a primitive type in the JSON string but got " + data['commitish']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      return true;
    }
  }]);
}();
/**
 * @member {String} branch
 */
ApiV1AppsIdBuildsPostRequest.prototype['branch'] = undefined;

/**
 * @member {String} commitish
 */
ApiV1AppsIdBuildsPostRequest.prototype['commitish'] = undefined;

/**
 * @member {String} description
 */
ApiV1AppsIdBuildsPostRequest.prototype['description'] = undefined;
var _default = exports["default"] = ApiV1AppsIdBuildsPostRequest;
