// build_io_api_v1/dist/model/App.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AppBuildpacksInner = _interopRequireDefault(require("./AppBuildpacksInner"));
var _AppFormationValue = _interopRequireDefault(require("./AppFormationValue"));
var _AppPipeline = _interopRequireDefault(require("./AppPipeline"));
var _AppTeam = _interopRequireDefault(require("./AppTeam"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Build.io API V1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The App model module.
 * @module model/App
 * @version v1
 */
var App = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>App</code>.
   * @alias module:model/App
   * @param id {String} 
   * @param name {String} 
   * @param team {module:model/AppTeam} 
   * @param stack {String} 
   * @param region {String} 
   */
  function App(id, name, team, stack, region) {
    _classCallCheck(this, App);
    App.initialize(this, id, name, team, stack, region);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(App, null, [{
    key: "initialize",
    value: function initialize(obj, id, name, team, stack, region) {
      obj['id'] = id;
      obj['name'] = name;
      obj['team'] = team;
      obj['stack'] = stack;
      obj['region'] = region;
    }

    /**
     * Constructs a <code>App</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/App} obj Optional instance to populate.
     * @return {module:model/App} The populated <code>App</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new App();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('team')) {
          obj['team'] = _AppTeam["default"].constructFromObject(data['team']);
        }
        if (data.hasOwnProperty('pipeline')) {
          obj['pipeline'] = _AppPipeline["default"].constructFromObject(data['pipeline']);
        }
        if (data.hasOwnProperty('dns_refreshed_at')) {
          obj['dns_refreshed_at'] = _ApiClient["default"].convertToType(data['dns_refreshed_at'], 'String');
        }
        if (data.hasOwnProperty('builds_count')) {
          obj['builds_count'] = _ApiClient["default"].convertToType(data['builds_count'], 'Number');
        }
        if (data.hasOwnProperty('deployments_count')) {
          obj['deployments_count'] = _ApiClient["default"].convertToType(data['deployments_count'], 'Number');
        }
        if (data.hasOwnProperty('slugs_count')) {
          obj['slugs_count'] = _ApiClient["default"].convertToType(data['slugs_count'], 'Number');
        }
        if (data.hasOwnProperty('current_build_id')) {
          obj['current_build_id'] = _ApiClient["default"].convertToType(data['current_build_id'], 'String');
        }
        if (data.hasOwnProperty('current_image_id')) {
          obj['current_image_id'] = _ApiClient["default"].convertToType(data['current_image_id'], 'String');
        }
        if (data.hasOwnProperty('current_deployment_id')) {
          obj['current_deployment_id'] = _ApiClient["default"].convertToType(data['current_deployment_id'], 'String');
        }
        if (data.hasOwnProperty('formation')) {
          obj['formation'] = _ApiClient["default"].convertToType(data['formation'], {
            'String': _AppFormationValue["default"]
          });
        }
        if (data.hasOwnProperty('stack')) {
          obj['stack'] = _ApiClient["default"].convertToType(data['stack'], 'String');
        }
        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }
        if (data.hasOwnProperty('buildpacks')) {
          obj['buildpacks'] = _ApiClient["default"].convertToType(data['buildpacks'], [_AppBuildpacksInner["default"]]);
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('pipeline_stage')) {
          obj['pipeline_stage'] = _ApiClient["default"].convertToType(data['pipeline_stage'], 'String');
        }
        if (data.hasOwnProperty('policy_allow_websockets')) {
          obj['policy_allow_websockets'] = _ApiClient["default"].convertToType(data['policy_allow_websockets'], 'Boolean');
        }
        if (data.hasOwnProperty('policy_response_timeout')) {
          obj['policy_response_timeout'] = _ApiClient["default"].convertToType(data['policy_response_timeout'], 'Number');
        }
        if (data.hasOwnProperty('policy_max_connections')) {
          obj['policy_max_connections'] = _ApiClient["default"].convertToType(data['policy_max_connections'], 'Number');
        }
        if (data.hasOwnProperty('policy_erosion_resistance_seconds')) {
          obj['policy_erosion_resistance_seconds'] = _ApiClient["default"].convertToType(data['policy_erosion_resistance_seconds'], 'Number');
        }
        if (data.hasOwnProperty('policy_share_process_namespace')) {
          obj['policy_share_process_namespace'] = _ApiClient["default"].convertToType(data['policy_share_process_namespace'], 'Boolean');
        }
        if (data.hasOwnProperty('policy_temporary_self_signed')) {
          obj['policy_temporary_self_signed'] = _ApiClient["default"].convertToType(data['policy_temporary_self_signed'], 'Boolean');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'String');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'String');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>App</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>App</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(App.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // validate the optional field `team`
      if (data['team']) {
        // data not null
        _AppTeam["default"].validateJSON(data['team']);
      }
      // validate the optional field `pipeline`
      if (data['pipeline']) {
        // data not null
        _AppPipeline["default"].validateJSON(data['pipeline']);
      }
      // ensure the json data is a string
      if (data['dns_refreshed_at'] && !(typeof data['dns_refreshed_at'] === 'string' || data['dns_refreshed_at'] instanceof String)) {
        throw new Error("Expected the field `dns_refreshed_at` to be a primitive type in the JSON string but got " + data['dns_refreshed_at']);
      }
      // ensure the json data is a string
      if (data['current_build_id'] && !(typeof data['current_build_id'] === 'string' || data['current_build_id'] instanceof String)) {
        throw new Error("Expected the field `current_build_id` to be a primitive type in the JSON string but got " + data['current_build_id']);
      }
      // ensure the json data is a string
      if (data['current_image_id'] && !(typeof data['current_image_id'] === 'string' || data['current_image_id'] instanceof String)) {
        throw new Error("Expected the field `current_image_id` to be a primitive type in the JSON string but got " + data['current_image_id']);
      }
      // ensure the json data is a string
      if (data['current_deployment_id'] && !(typeof data['current_deployment_id'] === 'string' || data['current_deployment_id'] instanceof String)) {
        throw new Error("Expected the field `current_deployment_id` to be a primitive type in the JSON string but got " + data['current_deployment_id']);
      }
      // ensure the json data is a string
      if (data['stack'] && !(typeof data['stack'] === 'string' || data['stack'] instanceof String)) {
        throw new Error("Expected the field `stack` to be a primitive type in the JSON string but got " + data['stack']);
      }
      // ensure the json data is a string
      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      }
      if (data['buildpacks']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['buildpacks'])) {
          throw new Error("Expected the field `buildpacks` to be an array in the JSON data but got " + data['buildpacks']);
        }
        // validate the optional field `buildpacks` (array)
        var _iterator2 = _createForOfIteratorHelper(data['buildpacks']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _AppBuildpacksInner["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['pipeline_stage'] && !(typeof data['pipeline_stage'] === 'string' || data['pipeline_stage'] instanceof String)) {
        throw new Error("Expected the field `pipeline_stage` to be a primitive type in the JSON string but got " + data['pipeline_stage']);
      }
      // ensure the json data is a string
      if (data['created_at'] && !(typeof data['created_at'] === 'string' || data['created_at'] instanceof String)) {
        throw new Error("Expected the field `created_at` to be a primitive type in the JSON string but got " + data['created_at']);
      }
      // ensure the json data is a string
      if (data['updated_at'] && !(typeof data['updated_at'] === 'string' || data['updated_at'] instanceof String)) {
        throw new Error("Expected the field `updated_at` to be a primitive type in the JSON string but got " + data['updated_at']);
      }
      // ensure the json data is a string
      if (data['deleted_at'] && !(typeof data['deleted_at'] === 'string' || data['deleted_at'] instanceof String)) {
        throw new Error("Expected the field `deleted_at` to be a primitive type in the JSON string but got " + data['deleted_at']);
      }
      return true;
    }
  }]);
}();
App.RequiredProperties = ["id", "name", "team", "stack", "region"];

/**
 * @member {String} id
 */
App.prototype['id'] = undefined;

/**
 * @member {String} name
 */
App.prototype['name'] = undefined;

/**
 * @member {module:model/AppTeam} team
 */
App.prototype['team'] = undefined;

/**
 * @member {module:model/AppPipeline} pipeline
 */
App.prototype['pipeline'] = undefined;

/**
 * @member {String} dns_refreshed_at
 */
App.prototype['dns_refreshed_at'] = undefined;

/**
 * @member {Number} builds_count
 */
App.prototype['builds_count'] = undefined;

/**
 * @member {Number} deployments_count
 */
App.prototype['deployments_count'] = undefined;

/**
 * @member {Number} slugs_count
 */
App.prototype['slugs_count'] = undefined;

/**
 * @member {String} current_build_id
 */
App.prototype['current_build_id'] = undefined;

/**
 * @member {String} current_image_id
 */
App.prototype['current_image_id'] = undefined;

/**
 * @member {String} current_deployment_id
 */
App.prototype['current_deployment_id'] = undefined;

/**
 * @member {Object.<String, module:model/AppFormationValue>} formation
 */
App.prototype['formation'] = undefined;

/**
 * @member {String} stack
 */
App.prototype['stack'] = undefined;

/**
 * @member {String} region
 */
App.prototype['region'] = undefined;

/**
 * @member {Array.<module:model/AppBuildpacksInner>} buildpacks
 */
App.prototype['buildpacks'] = undefined;

/**
 * @member {String} description
 */
App.prototype['description'] = undefined;

/**
 * @member {String} pipeline_stage
 */
App.prototype['pipeline_stage'] = undefined;

/**
 * @member {Boolean} policy_allow_websockets
 */
App.prototype['policy_allow_websockets'] = undefined;

/**
 * @member {Number} policy_response_timeout
 */
App.prototype['policy_response_timeout'] = undefined;

/**
 * @member {Number} policy_max_connections
 */
App.prototype['policy_max_connections'] = undefined;

/**
 * @member {Number} policy_erosion_resistance_seconds
 */
App.prototype['policy_erosion_resistance_seconds'] = undefined;

/**
 * @member {Boolean} policy_share_process_namespace
 */
App.prototype['policy_share_process_namespace'] = undefined;

/**
 * @member {Boolean} policy_temporary_self_signed
 */
App.prototype['policy_temporary_self_signed'] = undefined;

/**
 * @member {String} created_at
 */
App.prototype['created_at'] = undefined;

/**
 * @member {String} updated_at
 */
App.prototype['updated_at'] = undefined;

/**
 * @member {String} deleted_at
 */
App.prototype['deleted_at'] = undefined;
var _default = exports["default"] = App;
